gradio/components.py

```diff
@@ -335,6 +340,41 @@ def base64_to_temp_file_if_needed(
         self.temp_files.add(full_temp_file_path)
         return full_temp_file_path
 
+    def pil_to_temp_file(self, img: _Image.Image, dir: str, format=".png") -> str:
+        temp_dir = Path(dir) / self.hash_bytes(img.tobytes())
```

Comment:
Nit: the image bytes don't necessarily describe everything about the image in the presence of e.g. color profiles, metadata, etc.

Reply:
Added a test to make sure we're covered in this case